components/ledger/pkg/storage/sqlstorage/ledger/accounts.go

```diff
@@ -30,58 +29,34 @@ type Accounts struct {
 	Metadata map[string]string `bun:"metadata,type:jsonb,default:'{}'"`
 }
 
-type AccountsPaginationToken struct {
-	PageSize              uint                    `json:"pageSize"`
-	Offset                uint                    `json:"offset"`
-	AfterAddress          string                  `json:"after,omitempty"`
-	AddressRegexpFilter   string                  `json:"address,omitempty"`
-	MetadataFilter        map[string]string       `json:"metadata,omitempty"`
-	BalanceFilter         string                  `json:"balance,omitempty"`
-	BalanceOperatorFilter storage.BalanceOperator `json:"balanceOperator,omitempty"`
-}
-
-func (t AccountsPaginationToken) Encode() string {
-	return encodePaginationToken(t)
-}
-
-func (s *Store) buildAccountsQuery(ctx context.Context, p storage.AccountsQuery) (*bun.SelectQuery, AccountsPaginationToken) {
+func (s *Store) buildAccountsQuery(p storage.AccountsQuery) *bun.SelectQuery {
 	sb := s.schema.NewSelect(accountsTableName).Model((*Accounts)(nil))
-	t := AccountsPaginationToken{}
-
-	var (
-		address         = p.Filters.Address
-		metadata        = p.Filters.Metadata
-		balance         = p.Filters.Balance
-		balanceOperator = p.Filters.BalanceOperator
-	)
-
-	if address != "" {
-		sb.Where("address ~* ?", "^"+address+"$")
-		t.AddressRegexpFilter = address
+
+	if p.Payload.Address != "" {
```

Comment:
Payload is way less understandable than Filters, would it be possible to change it easily ?